spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindOverridableMethodCall.java

```diff
@@ -248,11 +248,9 @@ boolean checkDirectCase(XMethod caller, XMethod method, String message, int prio
     }
 
     private boolean shouldIgnoreCallInReadObject(XMethod method) {
-        // SER09-J-EX0: The readObject() method may invoke the overridable methods defaultReadObject() and readFields()
-        // in class java.io.ObjectInputStream
-        boolean inOIS = "java.io.ObjectInputStream".equals(method.getClassName());
-        String methodName = method.getName();
-        return inOIS && ("defaultReadObject".equals(methodName) || "readFields".equals(methodName));
+        // We're only interested in method calls on the object itself
+        // Calling ObjectInputStream.readInt() is not considered risky here because the object stream is potentially under the control of the attacker.
+        return !getClassContext().getClassDescriptor().getDottedClassName().equals(method.getClassName());
```

Comment:
IMO when calling inherited functions which are not present in the current class, should also give a warning.

Reply:
You're right and I have added a failing unit test to illustrate the case... I don't know how to check for inherited methods though!

Reply:
Ah I found it, maybe one day we should make that code more discoverable: it's not obvious that the static methods in `Hierarchy`, `Hierarchy2` or `Subtypes2` exist when starting from the `XMethod` interface